packages/debugger/test/data/fetch-storage.js

```diff
@@ -0,0 +1,130 @@
+import debugModule from "debug";
+const debug = debugModule("debugger:test:data:fetch-storage");
+
+import { assert } from "chai";
+import { promisify } from "util";
+
+import Ganache from "ganache";
+
+import { prepareContracts, testBlockGasLimit } from "../helpers";
+import Debugger from "lib/debugger";
+
+import * as Codec from "@truffle/codec";
+
+const __SIMPLE = `
+pragma solidity ^0.8.0;
+
+contract SimpleStorage {
+  uint public x;
+  constructor(uint initialX) {
+    //note: this is being included to get around
+    //https://github.com/trufflesuite/ganache/issues/3338 ;
+    //once that issue is fixed we should get rid of it
+    x = initialX;
+  }
+  function set(uint newX) public {
+    x = newX;
+  }
+}
+`;
+
+const sources = {
+  "SimpleStorage.sol": __SIMPLE
+};
+
+describe("Fetch storage option", function () {
+  let provider;
+
+  let abstractions;
+  let compilations;
+
+  async function mine() {
+    await promisify(provider.send.bind(provider))({
+      jsonrpc: "2.0",
+      method: "evm_mine",
+      id: Date.now(),
+      params: [Date.now()]
+    });
+  }
+
+  async function waitForSent(promiEvent) {
```

Comment:
a better name for this might be `waitForTransactionHash`, since that's what it returns?